feat: Document Request ID in debugging, tracing, and logging (#7441)
Kong/docs.konghq.com: app/_src/gateway/production/debug-request.md

The `X-Kong-Request-Debug-Token` is a token for authenticating the client and making the debug request to prevent abuse. Debug requests originating from loopback addresses don't require this header.

{% if_version gte:3.5.x %}
### X-Kong-Request-Id header

> The `X-Kong-Request-Id` is a unique identifier for each client request. This is enabled by default. You can use the unique request ID in this header to help you identify which request is causing the error when debugging. If {{site.base_gateway}} returns an error, the request ID will be included inside the response header as well as the JSON response generated by {{site.base_gateway}}. In addition, any {{site.base_gateway}} error log generated will also contains the same request ID, like `request_id: xxx`.

Reviewer: ```suggestion
The `X-Kong-Request-Id` header contains a unique identifier for each client request. This is enabled by default both upstream and downstream. This unique ID helps in matching specific requests to their corresponding error logs, which is useful for debugging. If {{site.base_gateway}} returns an error, the request ID will also be included in the response body generated by {{site.base_gateway}}. In addition, any {{site.base_gateway}} error log generated will contain the same request ID, with the format: `request_id: xxx`.
```

just a small suggestion, feel free to reword further (I removed JSON because that is not necessarily the only type of response body we return)

> The log line produced by the debug header as well as the debug response header contains the request ID. You can use this to search for the debug header output using a log viewer UI. This is especially useful when the debug output is too long to fit in the response header.

Reviewer 2: Can we add an example log line here?

Author: @lena-larionova There was a log example that included request id later on in the page in a different section, so I added a link to that instead of using the same/another example. Does that work?

Reviewer 2: That's great!

> The log line produced by the debug header as well as the debug response header contains the request ID which you can use for searching for the debug header output using a log viewer UI. For example, if the output is too long and can't fit into the response header, you can search for it inside of the error log instead.

Reviewer: ```suggestion
The log line produced by the debug header as well as the debug response header contains the request ID which you can use for searching for the debug header output using a log viewer UI. This is especially useful when the debug output is too long to fit in the response header.
```
